Kernel/TermIterators.cpp

```diff
@@ -439,22 +439,35 @@ Term* NonVariableNonTypeIterator::next()
 {
   Term* t = _stack.pop();
   TermList* ts;
-  _added = 0;  
-  Signature::Symbol* sym;
-  if (t->isLiteral()) {
-    sym = env.signature->getPredicate(t->functor());
-  } else{
-    sym = env.signature->getFunction(t->functor());
-  }
-  unsigned taArity; 
+  _added = 0;
+  unsigned taArity;
   unsigned arity;
-
-  if(t->isLiteral() && static_cast<Literal*>(t)->isEquality()){
+  if (t->isSpecial()) {
+    // This is a very incomplete support for special terms (which normally get eliminated during preprocessing).
+    // This satisfies the a use in AnswerLiteralManager (the synthesis version) where $ite-s may have creeped in.
+    // (We don't mind being iteration-incomplete there so skip the $ite-condition,
+    // which is a formula and would make things much more complicated here.)
+    // TODO decide: is it worth extending this properly (as usually, we won't encounter specail terms here)?
```

Comment:
Nit: "special", not "specail"

Reply:
...also actually: was the TermIterator used on special terms? I thought we only introduce the special if-then-elses at the very end, when constructing the program, and that we don't do anything with them except printing. (I can be wrong, will have more time to look at it tomorrow.)

Reply:
I think I did this because I saw synthesis break here after some (unifying) changes by me.